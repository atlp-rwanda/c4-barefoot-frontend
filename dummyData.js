export const user = [
  {
    "first_name":"M",
    "last_name":"JAckson",
    "email":"jackson.novelty@gmail.com",
    "occupation":"Programming",
    "username":"jacks",
    "bio":"Software Developer",
    "address":"kk 509 st",
    "language":"Kinyarwanda",
    "password":"12345678",
    "profile_picture":"https://res.cloudinary.com/mjackson/image/upload/v1609304524/Screen_Shot_2020-12-29_at_08.21.43_qe7pem.png"
  }
]

export const signupState = {
    requesting: false,
    errorOpen: false,
    erro: ""
  }
export const locationsPayload = [
    {
      id: "c6028e0d-ef88-4693-ab49-f37669891724",
      LocationName: "Kigali",
      country: "Rwanda",
      description: "The land of a thousand hills",
      link: "https://www.nationalgeographic.com/content/dam/travel/Guide-Pages/africa/downtown-kigali-rwanda.adapt.1900.1.jpg",
    }
  ]

export const accommodationsPayload = [
    {
      
        id: "0ce36391-2c08-3074-bddb-a4ea8cccbbc5",
        country: "Rwanda",
        city: "Kigali",
        title: "Marriot Hotel",
        description: "A serene environment for relaxation",
        photos: "https://cache.marriott.com/marriottassets/marriott/KGLMC/kglmc-exterior-0030-hor-feat.jpg",
    
    }
  ]

  export const accommodationState = {
    pending: false,
    accommodations: accommodationsPayload,
    error: null
  }

  export const locationState = {
    pending: false,
    locations: locationsPayload,
    error: null
  }

