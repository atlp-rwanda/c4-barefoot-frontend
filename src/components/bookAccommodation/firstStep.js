import React,{useEffect,useState} from "react";
import {Card,CardContent, CardActionArea, CardActions, CardMedia,Typography,Grid,Divider,Button,FormControlLabel,Checkbox} from '@material-ui/core';
import { Field, Form, Formik, FormikConfig,FormikValues} from 'formik';
import {getAccommodationsByLocation,selectAccommodation} from "../../redux/actions/fetchAccommodationByLocation";
import {getAccommodation,getAccommodations,getAccommodationAminity} from "../../redux/actions/fetchAccommodations";
import {getTemperature} from "../../redux/actions/getWeather";
import { connect } from 'react-redux';
import AccommodationCard from "../AccommodationCardWithReview";
import { makeStyles } from '@material-ui/core/styles';
import { Label, Place } from '@material-ui/icons'
import colors from '../colors';
import { Pagination } from '@material-ui/lab';
import Ratings from '../RatingStars';
import { Skeleton } from '@material-ui/lab';
import CloudIcon from '@material-ui/icons/Cloud';

const useStyles = makeStyles((theme) => ({
    media: {
      height: 440
    },
    checkbox:{
        display: 'block',
        position:'absolute',
        color: 'secondary',
        right: 0,
        fontSize:'25px'
    },
    cardActions:{
        display: 'flex',
        flexDirection:'column',
        alignItems:'flex-start',
    },
    separate:{
        marginLeft:theme.spacing(3),
        marginBottom:theme.spacing(3),
    },
    divider:{
        marginTop:theme.spacing(4)
    },
    separator:{
        marginBottom:theme.spacing(3),
        marginTop:theme.spacing(3),
        marginLeft:theme.spacing(2)
    },
    reviews:{
        fontSize: '14px',
        color: colors.primary100,
    },
    cardContent:{
      overflow: 'hidden'
    },
    container:{
        [theme.breakpoints.up("xs")]:{
            marginLeft:theme.spacing(3)
          },
          [theme.breakpoints.up("sm")]:{
            marginLeft:theme.spacing(9)
          }
        
    },
    btncontainer1:{
        display:"flex",
        justifyContent:"flex-start"
    },
    btncontainer2:{
        display:"flex",
        justifyContent:"flex-end"
    },
    titleText:{
      [theme.breakpoints.down('sm')]:{
        fontSize:'18px',
      }
    },
    item:{
        display:'block'
    }
  }));
function Home(props){
  const classes = useStyles();
    let temp= null;
    const [page, setPage] = useState(1);
    if(props.temp){
        temp=<Typography gutterBottom variant="h6"  className={classes.titleText} >
                <CloudIcon color="primary"/> {props.temp-273.15}                                  
            </Typography>
    }
    const display = props.accommodations.length ? 'none' : 'flex';
    const count=()=>{
        if(props.count % 6 ===0){
            return(props.count/6)
        }else{
            console.log("herre")
            return Math.trunc((props.count/6)+1)
        }
    }
    const handleChange = (event, value) => {
        props.getAccommodationsByLocation(props.accId,value)
        setPage(value)
      };
    return(
        <React.Fragment>
            <Card>
                <CardContent>
                    <Formik initialValues={{
                        from:null,
                        retrunDate:null
                    }}
                    onSubmit={values => {
                        props.nextStep();
                    }}>
                        <Form>
                        {(props.status ? 
                            <CardActionArea>
                            <Skeleton animation="wave" variant="rect" className={classes.media} />
                            <CardContent className={classes.cardContent} >
                                <Skeleton animation="wave" height={30} width="60%" />
                                <Skeleton animation="wave" height={10} width="80%" />
                            </CardContent>
                            <CardActions className={classes.cardActions}>
                                <Skeleton animation="wave" height={30} width="60%" />
                            </CardActions>
                            </CardActionArea>
                            :
                            <div >
                                <Grid container item xs={12}  className={classes.title,classes.container}  direction="column">
                                    <Typography variant="h6" style={{color: colors.primary100}} className={classes.separator}> 
                                        Choose Accommodation:
                                    </Typography>
                                    <Divider style={{width:'80%'}} variant='middle' />
                                                

                                </Grid>
                                <Grid container item xs={12} className={classes.container} direction='row'>
                                    {props.accommodations.length>0?(
                                        <Grid item>
                                        <Typography variant="subtitle1" style={{color: colors.primary100}} className={classes.separator}> 
                                            <Place color="secondary"/>{props.nation}
                                        </Typography>
                                    </Grid>
                                    ):(null)}
                                    
                                    <Grid  container item >
                                        {props.accommodations.map((accommodation) =>(
                                            <Grid item xs={8} sm={4} md={3} className={classes.insideGrid,classes.separate}>
                                                <AccommodationCard pending={props.status} accommodationn={accommodation} city={accommodation.city} {...props}  />
                                            </Grid>
                                        ))}
                                    </Grid>
                                    <Grid container item style={{display: display}} className={classes.notFound}>
                                        <Typography variant="h6" color="secondary" component="h6">No Accommodations found in this Locaton</Typography>
                                    </Grid>
                                    <Grid container item justify="center" style={{marginTop:'50px'}}>
                                    <Pagination count={count()} page={page} variant="outlined" color="primary" onChange={handleChange} />
                                    </Grid>
                                </Grid>
                                <div className={classes.btncontainer2}>
                                    <div className={classes.textCenter,classes.divider}>
                                        {props.selectedAccommodation?(<Button
                                            type='submit'
                                            variant='contained'
                                            color='primary'
                                            className={classes.button}
                                        >
                                            Next
                                        </Button>):(null)}
                                    </div>
                                </div>
                            </div>
                        )}
                        </Form>
                    </Formik>
                </CardContent>
            </Card>
        </React.Fragment>
    )
}

const mapStateToProps=state=>({
    accommodations:state.fetchAccommodations.accommodationsByLocation,
    accommodation:state.fetchAccommodations.accommodation,
    selectedAccommodation:state.fetchAccommodations.selectedAccommodation,
    count:state.fetchAccommodations.count,
    nation:state.fetchAccommodations.nation,
    accId:state.fetchAccommodations.accId,
    status:state.fetchAccommodations.pending,
    amenities:state.fetchAccommodations.amenities,
    temp:state.fetchAccommodations.temp
})
export default connect(mapStateToProps,{getAccommodationsByLocation,selectAccommodation,getAccommodation,getAccommodations,getAccommodationAminity,getTemperature}) (Home)