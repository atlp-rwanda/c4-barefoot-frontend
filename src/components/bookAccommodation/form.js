import React,{useEffect,useState} from "react";
import {Card,CardContent, CardActionArea, CardActions, CardMedia,Typography,Grid,Divider,Button,FormControlLabel,Checkbox} from '@material-ui/core';
import { Field, Form, Formik, FormikConfig,FormikValues} from 'formik';
import {getAccommodationsByLocation,selectAccommodation} from "../../redux/actions/fetchAccommodationByLocation";
import {getAccommodation,getAccommodations,getAccommodationAminity} from "../../redux/actions/fetchAccommodations";
import {getTemperature} from "../../redux/actions/getWeather";
import { connect } from 'react-redux';
import AccommodationCard from "../AccommodationCardWithReview";
import { makeStyles } from '@material-ui/core/styles';
import { Label, Place } from '@material-ui/icons'
import colors from '../colors';
import { Pagination } from '@material-ui/lab';
import Ratings from '../RatingStars';
import { Skeleton } from '@material-ui/lab';
import CloudIcon from '@material-ui/icons/Cloud';
import Book from "./index"

const useStyles = makeStyles((theme) => ({
    // root: {
    //   maxWidth: 345,
    //   height: 360,
    //   height: '100%',
    //   display: 'flex',
    //   flexDirection: 'column',
    //   justifyContent: 'space-between'
    // },
    media: {
      height: 440
    },
    checkbox:{
        display: 'block',
        position:'absolute',
        color: 'secondary',
        right: 0,
        fontSize:'25px'
    },
    cardActions:{
        display: 'flex',
        flexDirection:'column',
        alignItems:'flex-start',
    },
    separate:{
        // marginBottom:theme.spacing(3),
        marginLeft:theme.spacing(3)
    },
    divider:{
        // marginBottom:theme.spacing(3),
        marginTop:theme.spacing(4)
    },
    separator:{
        marginBottom:theme.spacing(3),
        marginTop:theme.spacing(3),
        marginLeft:theme.spacing(2)
    },
    reviews:{
        fontSize: '14px',
        color: colors.primary100,
    },
    cardContent:{
      overflow: 'hidden'
    },
    container:{
        marginLeft:theme.spacing(9),
        // width:'80%'
    },
    titleText:{
      [theme.breakpoints.down('sm')]:{
        fontSize:'18px',
      }
    },
    item:{
        display:'block'
    }
  }));
function Home(props){
    useEffect(() => {
    props.getAccommodationsByLocation(props.match.params.locationId);
    //console.log('here')
    props.getAccommodations();
  }, [])
  const classes = useStyles();
    let temp= null;
    if(props.temp){
        temp=<Typography gutterBottom variant="h6"  className={classes.titleText} >
                <CloudIcon color="primary"/> {props.temp-273.15}                                  
            </Typography>
    }
    const display = props.accommodations.length ? 'none' : 'flex';
    const count=()=>{
        if((props.count % 6) === 0){
            return (props.count / 6);
        }else if((props.count % 6) < 1){
            return 1;
        }else if((props.count % 6) >= 1){
            let pages= (props.count / 6)+1;
            return pages;
        }
    }

    const populateChecbox=()=>{
        let labels=[];
        let label;
        let count=0;
        if(props.amenities){
            const perm=props.amenities;
            for(const property in perm){
              label=
              <Grid item>
                <FormControlLabel
                        key={property}
                        control={<Checkbox checked={perm[property]} color={'primary'} name={property} id={count} disableRipple={true} />}
                        label={property}
                        color={colors.neutralBlack}
                />
              </Grid>
              count++;
              labels.push(label);
            }
            return labels;
          }else{
            return null;
        }
    }
    return(
        <Book/>
    )
}

const mapStateToProps=state=>({
    accommodations:state.fetchAccommodations.accommodationsByLocation,
    accommodation:state.fetchAccommodations.accommodation,
    count:state.fetchAccommodations.count,
    status:state.fetchAccommodations.pending,
    amenities:state.fetchAccommodations.amenities,
    temp:state.fetchAccommodations.temp
})
export default connect(mapStateToProps,{getAccommodationsByLocation,selectAccommodation,getAccommodation,getAccommodations,getAccommodationAminity,getTemperature}) (Home)