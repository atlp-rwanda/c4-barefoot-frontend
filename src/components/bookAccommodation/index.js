import FirstStep  from './firstStep';
import SecondStep from './secondStep';
import Confirm from './thirdStep';
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Stepper from '@material-ui/core/Stepper';
import Step from '@material-ui/core/Step';
import {Snackbar,Slide,Button} from '@material-ui/core';
import StepLabel from '@material-ui/core/StepLabel';
import Typography from '@material-ui/core/Typography';
import { connect } from 'react-redux'
import {clearBookSnackbar} from '../../redux/actions/bookAccommodationAction'
import { Avatar, Box } from '@material-ui/core';
import Skeleton from '@material-ui/lab/Skeleton';
import MuiAlert from '@material-ui/lab/Alert';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    textAlign: 'center',
    float: 'right',
    backgroundColor: '#fff',
  },
  rightSide: {
    border: '2px solid lightblue',
    padding: '1.5rem',
    '@media(max-width: 740px)' : {
      maxWidth: '450px',
    },
    '@media(max-width: 370px)' : {
      border: 'none',
      padding: 'unset'
    },
  },
  h3:{
    color: 'dodgerblue',
    margin: '1em',
    '@media(max-width: 740px)' : {
      fontSize: '15px'
    }
  },
  container: {
    display: 'grid',
    fontFamily: "'Poppins', sans-serif",
    '@media(max-width: 740px)' : {
      display: 'flex',
      flexDirection: 'column'
    },
    gridTemplateColumns: '1fr 1fr', 
    alignItems: 'center',
    justifyContent: 'center',
    padding: '1rem',
    boxSizing: 'border-box',
    marginBottom: '100px'
  },
  backButton: {
    marginRight: theme.spacing(1),
  },
  instructions: {
    marginTop: theme.spacing(1),
    marginBottom: theme.spacing(1),
  },
  textBox: {
    height: '20px',
    marginLeft: 'auto',
    marginRight: 'auto',
    marginBottom: '0.5rem',
    width: '100%',
    maxWidth: '200px'
  },
  box: {
    display: 'grid',
    alignItems: 'center'
  },
  stepperContainer: {
    display: 'flex',
    justifyContent: 'center',
    width: '400px',
    marginTop: '4px',
    
    '@media(max-width: 800px)' : {
      width: '330px'
    },
    '@media(max-width: 740px)' : {
      width: 'unset'
    },
  },
  stepperBox: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    minWidth: '100px',
    '@media(max-width: 360px)' : {
      justifyContent: 'start'
    }
  },
  stepperText: {
    height: '10px',
    width: '70%',
    marginTop: '5px'
  },
  container2:{
    width:'80%',
    [theme.breakpoints.up("xs")]:{
      marginLeft:theme.spacing(7)
    },
    [theme.breakpoints.up("sm")]:{
      marginLeft:theme.spacing(15)
    }
}
}));

function getSteps() {
  return ['Choose accommodation', 'View accommodation', 'confirm Booking'];
}

function getStepContent(loading, stepIndex, formData, setFormData, handleNext, handleBack) {
  switch (stepIndex) {
    case 0:
      return (
        <FirstStep
          formData={formData}
          setFormData={setFormData}
          nextStep={handleNext}
          loading = {loading}
        />
      );
    case 1:
      return (
        <SecondStep
          formData={formData}
          setFormData={setFormData}
          nextStep={handleNext}
          prevStep={handleBack}
          loading = {loading}
        />
      );
    case 2:
      return (
        <Confirm formData={formData} nextStep={handleNext} prevStep={handleBack} loading = {loading} />
      );
    default:
  }
}

const Book = (props)=> {
  const [formData, setFormData] = React.useState({
    From: '',
    To: ''
  });
  const classes = useStyles();
  const [activeStep, setActiveStep] = React.useState(0);
  const [loading, setLoading] = React.useState(true)
  const steps = getSteps();

  const closeBookSnackBarTimer = () => {
    props.clearBookSnackbar()
  }
  const TransitionUp = (props) => {
    return <Slide {...props} direction="up" />;
  }
  const handleNext = () => {
    setActiveStep((prevActiveStep) => prevActiveStep + 1);
  };

  const handleBack = () => {
    setActiveStep((prevActiveStep) => prevActiveStep - 1);
  };

  setTimeout(() => { setLoading(false)}, 1000);
  return (
    <div >
      <div >
        <div className={classes.rightSide}>    
          { loading ? (
            <Box className={classes.stepperContainer}>
              <Box className={classes.stepperBox}>
                <Skeleton variant="circle">
                  <Avatar/>
                </Skeleton>
                <Skeleton variant="rect" className={classes.stepperText}/>
                <Skeleton variant="rect" className={classes.stepperText}/>
              </Box>
              <Box className={classes.stepperBox}>
                <Skeleton variant="circle">
                  <Avatar/>
                </Skeleton>
                <Skeleton variant="rect" className={classes.stepperText}/>
                <Skeleton variant="rect" className={classes.stepperText}/>
              </Box>
              <Box className={classes.stepperBox}>
                <Skeleton variant="circle">
                  <Avatar/>
                </Skeleton>
                <Skeleton variant="rect" className={classes.stepperText}/>
                <Skeleton variant="rect" className={classes.stepperText}/>
              </Box>
            </Box>
          ) : (
            <Stepper activeStep={activeStep} alternativeLabel>
              {steps.map((label) => (
                <Step key={label}>
                  <StepLabel>{label}</StepLabel>
                </Step>
              ))}
            </Stepper>
          )}
          
          <div>
            {activeStep === steps.length ? (
              <div>
                  <Snackbar
                      open={props.snackBarMessage.open}
                      onClose={closeBookSnackBarTimer}
                      autoHideDuration={4000}
                      TransitionComponent={TransitionUp}
                  >
                  <MuiAlert
                      severity={props.snackBarMessage.severity}
                      variant='filled'
                      elevation={6}
                      >{props.snackBarMessage.message}
                  </MuiAlert>
                </Snackbar>
                <div className={classes.container2}>
                  <Typography className={classes.instructions}>You have Successfully Booked your Accommodation  <i class="fa fa-thumbs-up"></i></Typography>
                  <Button
                      color='primary'
                      variant='contained'
                  >
                      Back To your Requestes
                  </Button>
                </div>
              </div>
            ) : (
              <div>
                <Typography className={classes.instructions}>{getStepContent(loading, activeStep, formData, setFormData, handleNext, handleBack)}</Typography>
              </div>
            )}
          </div>
        </div>
      </div>
    </div>
  );
}

const mapStateToProps = state =>({
  snackBarMessage:state.bookAccommodations.snackBarMessage
})
export{Book}
export default connect(mapStateToProps,{clearBookSnackbar})(Book)

