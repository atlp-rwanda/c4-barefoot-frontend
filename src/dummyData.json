{
    "locations": [
      {
        "city": "capetown",
        "description": "This is one of best cities in Africa and you can also find good accomodations.",
        "image": "https://freewalkingtourscapetown.co.za/wp-content/uploads/2018/05/cropped-cape-town-header-pic-1.png"
      },
      {
        "city": "capetown",
        "description": "This is one of best cities in Africa and you can also find good accomodations.",
        "image": "https://freewalkingtourscapetown.co.za/wp-content/uploads/2018/05/cropped-cape-town-header-pic-1.png"
      },
      {
        "city": "capetown",
        "description": "This is one of best cities in Africa and you can also find good accomodations.",
        "image": "https://freewalkingtourscapetown.co.za/wp-content/uploads/2018/05/cropped-cape-town-header-pic-1.png"
      }
    ],
    "accommodations": [
      {
        "title": "Marriott Hotel",
        "location": "Kigali",
        "description": "This is one of best cities in Africa and you can also find good accomodations.",
        "image": "https://cache.marriott.com/marriottassets/marriott/KGLMC/kglmc-exterior-0030-hor-feat.jpg"
      },
      {
        "title": "Serena Hotel",
        "location": "Kigali",
        "description": "This is one of best cities in Africa and you can also find good accomodations.",
        "image": "https://cache.marriott.com/marriottassets/marriott/KGLMC/kglmc-exterior-0030-hor-feat.jpg"
      },
      {
        "title": "Marriott Hotel",
        "location": "Kigali",
        "description": "This is one of best cities in Africa and you can also find good accomodations.",
        "image": "https://cache.marriott.com/marriottassets/marriott/KGLMC/kglmc-exterior-0030-hor-feat.jpg"
      }
    ]
}
