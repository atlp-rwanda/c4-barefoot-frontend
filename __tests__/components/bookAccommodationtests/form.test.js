import React from 'react'
import Form from '../../../src/components/bookAccommodation/form'
import renderer from 'react-test-renderer'
import {Book} from '../../../src/components/bookAccommodation/index'
import { locationState, accommodationState,bookAccommodationState } from '../../../dummyData'
import { shallow , mount} from "enzyme";
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux'
import thunk from 'redux-thunk';


const middlewares = [thunk]
const mockStore = configureStore(middlewares);
let store;

store = mockStore({
  fetchAccommodations:  accommodationState,
  bookAccommodations:bookAccommodationState
})

const props = {
  getAccommodations: jest.fn(),
  getAccommodationsByLocation: jest.fn()
}

describe('Book accommodation from', () => {
  
  it('Render book accommodation from', () => {
    
    const renderedComponent = renderer.create(<Provider store={store}>
        <Form {...props}/>
    </Provider>).toJSON()
    expect(renderedComponent).toMatchSnapshot();
    
  })
})

