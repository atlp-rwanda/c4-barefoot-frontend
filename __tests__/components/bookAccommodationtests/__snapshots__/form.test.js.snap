// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book accommodation from Render book accommodation from 1`] = `
<div>
  <div>
    <div
      className="makeStyles-rightSide-2"
    >
      <div
        className="MuiBox-root MuiBox-root-14 makeStyles-stepperContainer-9"
      >
        <div
          className="MuiBox-root MuiBox-root-15 makeStyles-stepperBox-10"
        >
          <span
            className="MuiSkeleton-root MuiSkeleton-circle MuiSkeleton-withChildren MuiSkeleton-fitContent MuiSkeleton-heightAuto MuiSkeleton-pulse"
            style={
              Object {
                "height": undefined,
                "width": undefined,
              }
            }
          >
            <div
              className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiAvatar-fallback"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
          </span>
          <span
            className="MuiSkeleton-root MuiSkeleton-rect makeStyles-stepperText-11 MuiSkeleton-pulse"
            style={
              Object {
                "height": undefined,
                "width": undefined,
              }
            }
          />
          <span
            className="MuiSkeleton-root MuiSkeleton-rect makeStyles-stepperText-11 MuiSkeleton-pulse"
            style={
              Object {
                "height": undefined,
                "width": undefined,
              }
            }
          />
        </div>
        <div
          className="MuiBox-root MuiBox-root-16 makeStyles-stepperBox-10"
        >
          <span
            className="MuiSkeleton-root MuiSkeleton-circle MuiSkeleton-withChildren MuiSkeleton-fitContent MuiSkeleton-heightAuto MuiSkeleton-pulse"
            style={
              Object {
                "height": undefined,
                "width": undefined,
              }
            }
          >
            <div
              className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiAvatar-fallback"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
          </span>
          <span
            className="MuiSkeleton-root MuiSkeleton-rect makeStyles-stepperText-11 MuiSkeleton-pulse"
            style={
              Object {
                "height": undefined,
                "width": undefined,
              }
            }
          />
          <span
            className="MuiSkeleton-root MuiSkeleton-rect makeStyles-stepperText-11 MuiSkeleton-pulse"
            style={
              Object {
                "height": undefined,
                "width": undefined,
              }
            }
          />
        </div>
        <div
          className="MuiBox-root MuiBox-root-17 makeStyles-stepperBox-10"
        >
          <span
            className="MuiSkeleton-root MuiSkeleton-circle MuiSkeleton-withChildren MuiSkeleton-fitContent MuiSkeleton-heightAuto MuiSkeleton-pulse"
            style={
              Object {
                "height": undefined,
                "width": undefined,
              }
            }
          >
            <div
              className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiAvatar-fallback"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
          </span>
          <span
            className="MuiSkeleton-root MuiSkeleton-rect makeStyles-stepperText-11 MuiSkeleton-pulse"
            style={
              Object {
                "height": undefined,
                "width": undefined,
              }
            }
          />
          <span
            className="MuiSkeleton-root MuiSkeleton-rect makeStyles-stepperText-11 MuiSkeleton-pulse"
            style={
              Object {
                "height": undefined,
                "width": undefined,
              }
            }
          />
        </div>
      </div>
      <div>
        <div>
          <p
            className="MuiTypography-root makeStyles-instructions-6 MuiTypography-body1"
          >
            <div
              className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                className="MuiCardContent-root"
              >
                <form
                  action="#"
                  form-data="form-1"
                  onReset={[Function]}
                  onSubmit={[Function]}
                >
                  <button
                    className="MuiButtonBase-root MuiCardActionArea-root"
                    disabled={false}
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiSkeleton-root MuiSkeleton-rect makeStyles-media-18 MuiSkeleton-wave"
                      style={
                        Object {
                          "height": undefined,
                          "width": undefined,
                        }
                      }
                    />
                    <div
                      className="MuiCardContent-root makeStyles-cardContent-25"
                    >
                      <span
                        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
                        style={
                          Object {
                            "height": 30,
                            "width": "60%",
                          }
                        }
                      />
                      <span
                        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
                        style={
                          Object {
                            "height": 10,
                            "width": "80%",
                          }
                        }
                      />
                    </div>
                    <div
                      className="MuiCardActions-root makeStyles-cardActions-20 MuiCardActions-spacing"
                    >
                      <span
                        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
                        style={
                          Object {
                            "height": 30,
                            "width": "60%",
                          }
                        }
                      />
                    </div>
                    <span
                      className="MuiCardActionArea-focusHighlight"
                    />
                  </button>
                </form>
              </div>
            </div>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
